sound/soc/sof/ipc4-topology.c

```diff
@@ -1480,6 +1502,25 @@ static int sof_ipc4_get_valid_bits(struct snd_sof_dev *sdev, struct snd_pcm_hw_p
 	}
 }
 
+static int sof_ipc4_get_sample_type(struct snd_sof_dev *sdev, struct snd_pcm_hw_params *params)
+{
+	switch (params_format(params)) {
+	case SNDRV_PCM_FORMAT_A_LAW:
+		return SOF_IPC4_TYPE_A_LAW;
+	case SNDRV_PCM_FORMAT_MU_LAW:
+		return SOF_IPC4_TYPE_MU_LAW;
+	case SNDRV_PCM_FORMAT_U8:
+		return SOF_IPC4_TYPE_UNSIGNED_INTEGER;
+	case SNDRV_PCM_FORMAT_S16_LE:
+	case SNDRV_PCM_FORMAT_S24_LE:
+	case SNDRV_PCM_FORMAT_S32_LE:
+		return SOF_IPC4_TYPE_LSB_INTEGER;
+	default:
+		dev_err(sdev->dev, "invalid pcm sample type %d\n", params_format(params));
+		return -EINVAL;
```

Comment:
HDMI supports IEC958_SUBFRAME_LE, that should be not an error.

Reply:
@ujfalusi This seems to be called only for capture. I was never able to get SNDRV_PCM_FORMAT_IEC958_SUBFRAME_LE. What do you think, should I remove the added case since it doesn't get executed?

As alternative way, this function could return by default SOF_IPC4_TYPE_LSB_INTEGER and only warn if ALSA PCM format is unknown. I don't know really what to do other than this conversion from ALSA to IPC4 needs to be done for 8-bit and next to-be-added float sample type.